WordPress/Classes/Services/RawBlockEditorSettingsService.swift

```diff
@@ -4,68 +4,81 @@ import WordPressShared
 
 class RawBlockEditorSettingsService {
     private let blog: Blog
-    private let remoteAPI: WordPressOrgRestApi
     private var isRefreshing: Bool = false
+    private var refreshTask: Task<[String: Any], Error>?
 
-    init?(blog: Blog) {
-        guard let remoteAPI = WordPressOrgRestApi(blog: blog) else {
-            return nil
-        }
-
+    init(blog: Blog) {
         self.blog = blog
-        self.remoteAPI = remoteAPI
+    }
+
+    private static var services: [TaggedManagedObjectID<Blog>: RawBlockEditorSettingsService] = [:]
+
+    @MainActor
+    static func getService(forBlog blog: Blog) -> RawBlockEditorSettingsService {
```

Comment:
The previous implementation with `isRefreshing` was ineffective because there were multiple instances of `RawBlockEditorSettingsService`: one for My Site and one for each editor.

Reply:
That makes sense now that you call attention to it. I like the new design with static methods for retrieving the individual services.